src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java

```diff
@@ -28,6 +28,10 @@ public NegotiateResponse(String negotiatePayload) {
                     case "error":
                         this.error = reader.nextString();
                         break;
+                    case "ProtocolVersion":
+                        this.error = "Detected a connection attempt to an ASP.NET SignalR Server. This client only supports connecting to an ASP.NET Core SignalR Server. See https://aka.ms/signalr-core-differences for details.";
+                        reader.skipValue();
```

Comment:
should we just return immediately here?

Reply:
That would be reasonable. There's no need to finish parsing, the rest is garbage anyway.

Reply:
👍